### PR TITLE
No burrow surfacing on lifeboats/unburrowable areas

### PR DESCRIPTION
**Pull request by hislittlecuzingames**
# About the pull request

This is to prevent burrowers from surfacing in areas marked as not tunnleable. 
IE the lifeboats in an evac of the almeyer. 

# Explain why it's good for the game

I think surfacing was an oversight of the original burrow ability. 
This will stop goofy burrows that are nonsensical like a burrower emerging inside a lifeboat during a hijack. 
HOWEVER, it doesn't stop burrowers from burrowing on the almeyer. Only /area/foobar that are explicitly tagged as flags_area = AREA_NOTUNNEL cannot be emerged from now. before it was just preventing burrowing down into these tiles. 

# Testing Photographs and Procedure

<details>
Map changes were not included. Only code changes were made.
Chance's claim is just my default map for some reason, so I used its areas and edited them temporarily for testing. 

https://youtu.be/EwlO7mJ017s

</details>

# Changelog

:cl:
fix: fixed burrower ability. Disallowed ability to surface in areas marked as not tunnel-able.
/:cl:

### CODE REVIEW

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by hislittlecuzingames** [comment]
> This pull request has conflicts, please resolve those before we can evaluate the pull request.

Yeah I gotta fix my base branch-has some artifacts. 

SHould be good now.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Comment by hislittlecuzingames** [comment]
> Conflicts have been resolved. A maintainer will review the pull request shortly.

oh you're a robot

**Comment by hislittlecuzingames** [comment]
I tried some frivilous code changes to appease Lint, but it didn't help, so I put it back to  the way I wrote it the last time I verified without a doubt that the code behaved properly.

File: code/modules/mob/living/carbon/xenomorph/abilities/burrower/burrower_powers.dm [hunk 1]
```diff
@@ -96,6 +96,15 @@
 
 	if(!HAS_TRAIT(src, TRAIT_ABILITY_BURROWED))
 		to_chat(src, SPAN_NOTICE("We must be burrowed to do this."))
+
+	var/area/current_area = get_area(T)
+	if(current_area.flags_area & AREA_NOTUNNEL)
+		to_chat(src, SPAN_XENOWARNING("There's no way to surface here!!"))
+		return
+
+	if(!burrow)
```
Comment (hislittlecuzingames) [inline code/modules/mob/living/carbon/xenomorph/abilities/burrower/burrower_powers.dm:105]: I didn't write this... am I going insane? 

I'm going to look at it tomorrow, but I'm quite certain this was already there.

Comment (hislittlecuzingames) [reply code/modules/mob/living/carbon/xenomorph/abilities/burrower/burrower_powers.dm:105]: This is weird that it wrote like 105/106/107 in my pr... my local git doesn't say I wrote that so, that's bizzare.

Comment (hislittlecuzingames) [reply code/modules/mob/living/carbon/xenomorph/abilities/burrower/burrower_powers.dm:105]: Lint is saying `burrow` is undefined, but it is defined in the `Xenomorph.dm` class and there is a macro in `Burrower.dm` in the `base_action` list. 

Is lint just unable to recognise these things correctly?

**Comment by hislittlecuzingames** [comment]
Most of the changes were to fix the small grammer stuff.

The fix is very small.

File: code/modules/mob/living/carbon/xenomorph/abilities/burrower/burrower_powers.dm [hunk 2]
```diff
@@ -98,6 +98,11 @@
 		to_chat(src, SPAN_NOTICE("We must be burrowed to do this."))
 		return
 
+	var/area/current_area = get_area(T)
+	if(current_area.flags_area & AREA_NOTUNNEL)
+		to_chat(src, SPAN_XENOWARNING("There's no way to surface here!!"))
+		return
+
```
Comment (Drulikar) [inline code/modules/mob/living/carbon/xenomorph/abilities/burrower/burrower_powers.dm:105]: This is unnecessary; its already done further down at line 129/134 so should revert this change.

The difference is you spend some time *trying* to tunnel there and find out you can't - which is fine.

File: code/game/area/shuttles.dm [hunk 3]
```diff
@@ -82,3 +82,4 @@
 	icon = 'icons/turf/area_almayer.dmi'
 	icon_state = "lifeboat"
 	flags_atom = AREA_NOTUNNEL
+	flags_area = AREA_NOTUNNEL
```
Comment (Drulikar) [inline code/game/area/shuttles.dm:85]: flags_atom shouldn't have been this value.
```suggestion
	flags_area = AREA_NOTUNNEL
```

**Review by Drulikar — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/xenomorph/abilities/burrower/burrower_powers.dm (hunk 2 above)
Comment (hislittlecuzingames) [reply code/modules/mob/living/carbon/xenomorph/abilities/burrower/burrower_powers.dm:105]: Without that code, it wouldn't work. 
It lets them surface.

**Comment by hislittlecuzingames** [comment]
Going to re-test it another day since I committed suggested changes. 

Also going to verify my comment that I needed the code stated to be unnecessary.

File: code/modules/mob/living/carbon/xenomorph/abilities/burrower/burrower_powers.dm (hunk 2 above)
Comment (Drulikar) [reply code/modules/mob/living/carbon/xenomorph/abilities/burrower/burrower_powers.dm:105]: Did you test this? Because I can comment it out and it still is checked because its a duplicate check as I said.

So either you need to more clearly specify what conditions require this duplicate check; but I don't currently see any situation this isn't simply redundant.

image

**Review by Drulikar — CHANGES_REQUESTED** [review]

**Comment by hislittlecuzingames** [comment]
I said I would test at a later date. 
https://github.com/cmss13-devs/cmss13/pull/5603#issuecomment-1949657319
